rust_src/src/obarray.rs

```diff
@@ -99,13 +92,60 @@ impl LispObarrayRef {
         } else if LispObject::from_raw(unsafe { globals.f_Vpurify_flag }).is_not_nil() {
             // When Emacs is running lisp code to dump to an executable, make
             // use of pure storage.
-            LispObject::from_raw(unsafe {
-                intern_driver(Fpurecopy(string.to_raw()), self.0.to_raw(), tem.to_raw())
-            })
+            LispObject::from_raw(intern_driver(
+                unsafe { Fpurecopy(string.to_raw()) },
+                self.0.to_raw(),
+                tem.to_raw(),
+            ))
         } else {
-            LispObject::from_raw(unsafe {
-                intern_driver(string.to_raw(), self.0.to_raw(), tem.to_raw())
-            })
+            LispObject::from_raw(intern_driver(
+                string.to_raw(),
+                self.0.to_raw(),
+                tem.to_raw(),
+            ))
+        }
+    }
+}
+
+/// Get an error if OBARRAY is not an obarray.
+/// If it is one, return it.
+#[no_mangle]
+pub extern "C" fn check_obarray(obarray: Lisp_Object) -> Lisp_Object {
```

Comment:
Why is this explicitly `extern "C"`? Let the `lisp_fn` macro generate the `extern "C" version of the function for you.

Reply:
Because it is needed in C land too. It is not called from elisp.

Reply:
This is a `no_mangle` not a `lisp_fn`.

Reply:
Oh right, it's not called from elisp. How many C callers are left?

Reply:
More than 11.

Reply:
Sorry that is the original tree. It is 7 after this PR.

Reply:
That's not many, but it's enough that I won't make you guys port them all before merging this in.

Reply:
Good because the others are not as easy :-) The rest of this file gets deep into s-expression parsing, memory allocation calls, etc.

Reply:
gotta tackle that sooner or later though :)